### PR TITLE
[fix](partial update) update error code when failed to fill the missing fields

### PR DESCRIPTION
**Pull request by zhannngchen**
## Proposed changes

Issue Number: close #xxx

1. InternalError is not clear for such error, use InvalidSchema Error instead
2. avoid some useless stacktrace on InternalError when load failed

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhannngchen** [comment]
run buildall

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Review by bobhan1 — APPROVED** [review]

File: be/src/olap/rowset/segment_v2/segment_writer.cpp [hunk 1]
```diff
@@ -471,7 +471,7 @@ Status SegmentWriter::append_block_with_partial_content(const vectorized::Block*
             }
 
             if (!_opts.rowset_ctx->partial_update_info->can_insert_new_rows_in_partial_update) {
-                return Status::InternalError(
+                return Status::Error<INVALID_SCHEMA, false>(
```
Comment (gavinchou) [inline be/src/olap/rowset/segment_v2/segment_writer.cpp:474]: We may attach some context to the msg: a transaction id or load id

Comment (zhannngchen) [reply be/src/olap/rowset/segment_v2/segment_writer.cpp:482]: should attach the load_id at _flush_memtable

**Comment by zhannngchen** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Review by bobhan1 — APPROVED** [review]

**Review by gavinchou — APPROVED** [review]

**Comment by zhannngchen** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**